sale_purchase_force_vendor/views/sale_order_view.xml

```diff
@@ -26,7 +26,7 @@
                     <field
                         name="vendor_id"
                         domain="vendor_id_domain"
-                        readonly="'state' in ('done', 'cancel')"
+                        readonly="'state' in ('sale', 'cancel') or parent.locked"
```

Comment:
Uhm, thinking twice, it was incorrect before, as you can't change the vendor once the quotation is confirmed. It's enough with the state.